### ISSUE TITLE
Support to weaken strong ETags when compressing/decompressing

### ISSUE THREAD

**Issue by httpdigest** [envoyproxy/envoy#35152]
*Title*: *Support to weaken strong ETags when compressing/decompressing*

*Description*:
Currently, when using the compressor filter, there are basically two options to handle (strong) ETags:
- when setting `disable_on_etag_header: true` then do not compress a response
- when setting `disable_on_etag_header: false` then throw away the strong ETag (only keep weak ETags)

But one other option, that is popular in other HTTP proxies/caches, for example Varnish HTTP Cache, is:
- weaken any strong ETag by prepending it with the "W/" prefix when compressing a response before sending it to the client

When searching the web, there are _numerous_ stackoverflow, bugzilla (or other bug tracking tools) and blog posts about how to handle ETags and how to do it in the least surprising ways (surprising for any developer or server/platform operator).

Now, since we were using Varnish and I was familiar with the "weaken any strong ETag when compressing" behaviour, it caught me by surprise that the Envoy compressor does _not_ provide such an option (only the above two mentioned options).

My current use-case is to compress/decompress the traffic between each two Istio sidecars when the backends do not:
- advertize in client requests that they support compressiong via Accept-Encoding
- compress the response themselves when the client did advertize support for it and then adding a Content-Encoding response header

Would it be possible to provide a way to weaken any strong ETag with Envoy's compressor here?

**Comment by nezdolik** [comment]
cc @kbaichoo @mattklein123

**Comment by httpdigest** [comment]
Not stale.

**Comment by KBaichoo** [comment]
Sorry, this fell off my radar @httpdigest. This sounds reasonable to me if it's an additional option. e.g. we have several modes of how etag is supported and the behavior you described being one of them.

**Comment by httpdigest** [comment]
Yeah, having the option to configure it like that, would be very much preferable for us.
